lib/Widget/Provider/DataProviderInterface.php

```diff
@@ -76,6 +76,12 @@ public function getDisplayLongitude(): float;
      */
     public function getGuzzleClient(array $requestOptions = []): \GuzzleHttp\Client;
 
+    /**
+     * Get Stash pool interface
+     *  User it if any data needs to be cached inside the Data Provider
+     */
+    public function getPool() : \Stash\Interfaces\PoolInterface;
```

Comment:
This should be inside whatever is using the data provider - not the actual data provider.